container_ci_suite/engines/container.py

```diff
@@ -21,69 +21,508 @@
 
-    @staticmethod
-    def docker_get_user_id(src_image, user):
-        return PodmanCLIWrapper.docker_run_command(
-            f"--rm {src_image} bash -c 'id -u {user} 2>/dev/null"
+    # Replacement for ct_s2i_build_as_df
+    def s2i_build_as_df(self, app_path: str, s2i_args: str, src_image: str, dst_image: str):
+        named_tmp_dir = mkdtemp()
+        tmp_dir = Path(named_tmp_dir)
+        if tmp_dir.exists():
+            logger.debug("Temporary Directory exists.")
+        else:
+            logger.debug("Temporary directory not exists.")
+        ntf = mktemp(dir=str(tmp_dir), prefix="Dockerfile.")
+        df_name = Path(ntf)
+        df_content = self.s2i_create_df(
+            tmp_dir=tmp_dir,
+            app_path=app_path,
+            s2i_args=s2i_args,
+            src_image=src_image,
+            dst_image=dst_image,
         )
+        with open(df_name, mode="w") as f:
+            f.writelines(df_content)
+        mount_options = get_mount_options_from_s2i_args(s2i_args=s2i_args)
+        # Run the build and tag the result
+        PodmanCLIWrapper.run_docker_command(
+            f"build {mount_options} -f {df_name} --no-cache=true -t {dst_image}"
+        )
+        return ContainerImage(image_name=dst_image)
 
-    @staticmethod
-    def docker_pull_image(image_name: str, loops: int = 10) -> bool:
-        """
-        Function checks if image_name is present in system.
-        In case it isn't, try to pull it for specific count of loops
-        Default is 10.
-        """
-        if PodmanCLIWrapper.docker_image_exists(image_name=image_name):
-            print("Pulled image already exists.")
-            return True
-        for loop in range(loops):
-            ret_val = PodmanCLIWrapper.run_docker_command(
-                cmd=f"pull {image_name}", return_output=False
+    # Replacement for ct_s2i_build_as_df_build_args
+    def s2i_create_df(
+        self, tmp_dir: Path, app_path: str, s2i_args: str, src_image, dst_image: str
+    ) -> List[str]:
+        real_app_path = app_path.replace("file://", "")
+        df_content: List = []
+        local_scripts: str = "upload/scripts"
+        local_app: str = "upload/src"
+        os.chdir(tmp_dir)
+        if not PodmanCLIWrapper.docker_image_exists(src_image):
+            if "pull-policy=never" not in s2i_args:
+                PodmanCLIWrapper.run_docker_command(f"pull {src_image}")
+
+        user = PodmanCLIWrapper.docker_inspect(
+            field="{{.Config.User}}", src_image=src_image
+        )
+        if not user:
+            user = "0"
+
+        assert int(user)
+        user_id = PodmanCLIWrapper.docker_get_user_id(src_image=src_image, user=user)
+        if not user_id:
+            logger.error(f"id of user {user} not found inside image {src_image}.")
+            logger.error("Terminating s2i build.")
+            return None
+        else:
+            user_id = user
```

Comment:
The `else` statement is redundant. Also, `user` variable can (in my testing) contain a newline at the end, which broke the dockerfile.

I'd remove `else` statement and use `.strip()` in `docker_get_user_id` function in `podman_wrapper.py`.

_Strip function probably should be in the `docker_inspect` function as well. Can't think of a reason why we would want output with trailing/leading whitespaces._

Reply:
Good point. 'else' is removed by this commit https://github.com/sclorg/container-ci-suite/pull/93/commits/9a165e38ebd34b3d2826faaebd8710e99ab03b1e